### PR TITLE
Ardupilot downstream changes (sync upto 2019-12-10)

### PR DESCRIPTION
**Pull request by amilcarlucas**

### CODE REVIEW

**Comment by amilcarlucas** [comment]
Please rebase, but do not squash

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -2642,6 +2649,36 @@
       <entry value="10" name="SERIAL_CONTROL_DEV_SHELL">
         <description>system shell</description>
       </entry>
+      <entry value="100" name="SERIAL_CONTROL_SERIAL0">
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml]: @julianoes Can you please assess the implications of adding specific serial items here. I know nothing about this.

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -990,6 +990,9 @@
       <entry value="4" name="MAV_MOUNT_MODE_GPS_POINT">
         <description>Load neutral position and start to point to Lat,Lon,Alt</description>
       </entry>
+      <entry value="5" name="MAV_MOUNT_MODE_SYSID_TARGET">
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml]: I've been discussing this and MAV_CMD_DO_SET_ROI_SYSID with Peter. Merged in https://github.com/mavlink/mavlink/pull/1267. Can you fix up downstream?

Comment (amilcarlucas) [reply message_definitions/v1.0/common.xml]: I rebased this one. Will port dowstream now

Comment (hamishwillee) [reply message_definitions/v1.0/common.xml]: Thanks. There is some formatting issue with the xml too (see checks)

Comment (amilcarlucas) [reply message_definitions/v1.0/common.xml]: Downstream ported and formatting fixed

Comment (hamishwillee) [reply message_definitions/v1.0/common.xml]: @amilcarlucas 
> attribute 'units': 'kPa' is not a valid value of the atomic type 'SI_Unit'.

I've created https://github.com/mavlink/mavlink/pull/1282 to pull in the updates. Any reason not to merge that?

**Comment by amilcarlucas** [comment]
OK, now all the CI tests pass.

File: message_definitions/v1.0/common.xml (hunk 2 above)
Comment (amilcarlucas) [reply message_definitions/v1.0/common.xml]: This one has everything in a single PR

File: message_definitions/v1.0/ardupilotmega.xml [hunk 3]
```diff
@@ -111,6 +111,16 @@
         <param index="6">Empty.</param>
         <param index="7">Empty.</param>
       </entry>
+      <entry value="42006" name="MAV_CMD_FIXED_MAG_CAL_YAW">
+        <description>Magnetometer calibration based on provided known yaw. This allows for fast calibration using WMM field tables in the vehicle, given only the known yaw of the vehicle. If Latitude and longitude are both zero then use the current vehicle location.</description>
+        <param index="1" label="Yaw" units="deg">Yaw of vehicle in earth frame.</param>
+        <param index="2" label="CompassMask">CompassMask, 0 for all.</param>
+        <param index="3" label="Lattitude" units="deg">Latitude.</param>
```
Comment (julianoes) [inline message_definitions/v1.0/ardupilotmega.xml]:
```suggestion
        <param index="3" label="Latitude" units="deg">Latitude.</param>
```

File: message_definitions/v1.0/common.xml (hunk 1 above)
Comment (julianoes) [reply message_definitions/v1.0/common.xml]: I have no idea what this is about or how it works.

File: message_definitions/v1.0/ardupilotmega.xml (hunk 3 above)
Comment (amilcarlucas) [reply message_definitions/v1.0/ardupilotmega.xml]: Good catch

**Comment by amilcarlucas** [comment]
All checks pass now

File: message_definitions/v1.0/common.xml (hunk 1 above)
Comment (hamishwillee) [reply message_definitions/v1.0/common.xml]: @amilcarlucas Can you explain what this is for and why we need extra? I'm in no way opposed, just ignorant.

**Pull request merged**